salt/tests/unit/modules/files/test_metalk8s_kubernetes_utils.yaml

```diff
@@ -0,0 +1,89 @@
+    }
+    result: ['my-config-option-kubeconfig.conf', 'my-config-option-mycontext']
+
+  # 4. get kubeconfig partly from kwargs and salt config options
+  - kwargs: {"context": "my-kwargs-mycontext"}
+    config_options: {"kubernetes.kubeconfig": "my-config-option-kubeconfig.conf"}
+    result: ['my-config-option-kubeconfig.conf', 'my-kwargs-mycontext']
+  # 5. get kubeconfig partly from kwargs and api-server pillars
+  - kwargs: {"kubeconfig": "my-kwargs-kubeconfig.conf"}
+    api_server_pillar: {"context": "my-pillar-context"}
+    result: ['my-kwargs-kubeconfig.conf', 'my-pillar-context']
+  # 6. get kubeconfig partly from salt config options and context result is None
+  - config_options: {"kubernetes.kubeconfig": "my-config-option-kubeconfig.conf"}
+    result: ['my-config-option-kubeconfig.conf', null]
+  # 7. get kubeconfig partly from api-server pillar and context result is None
+  - api_server_pillar: {"kubeconfig": "my-pillar-kubeconfig.conf"}
+    result: ['my-pillar-kubeconfig.conf', null]
+  # 8. get kubeconfig partly from kwargs and context result is None
+  - kwargs: {"kubeconfig": "my-kwargs-kubeconfig.conf"}
+    result: ['my-kwargs-kubeconfig.conf', null]
+  # 9. test that kwargs overrides pillar and salt config option
+  - kwargs: {"kubeconfig": "my-kwargs-kubeconfig.conf", "context": "my-kwargs-mycontext"}
+    api_server_pillar: {
+      "kubeconfig": "my-pillar-kubeconfig.conf",
+      "context": "my-pillar-context"
+    }
+    config_options: {"kubernetes.kubeconfig": "my-config-option-kubeconfig.conf"}
+    result: ['my-kwargs-kubeconfig.conf', 'my-kwargs-mycontext']
+  # 10. test that kubeconfig is defined in the lowest level and context is being overridden
+  - api_server_pillar: {"context": "my-pillar-context"}
+    config_options: {
+      "kubernetes.kubeconfig": "my-config-option-kubeconfig.conf",
+      "kubernetes.context": "my-config-option-mycontext"
+    }
+    result: ['my-config-option-kubeconfig.conf', 'my-pillar-context']
+
+read_and_render_yaml_file:
+  # 1. read and render a valid yaml file
+  - source:
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: kube-apiserver
+    result:
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: "kube-apiserver"
+  #. 2. Raise error when no source file is specified
+  - source: null
+    raises: True
+    result: 'Source file my-source-file not found'
+  # 3. Raise error when a valid template is given but salt is unable to render the file
+  - source:
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: kube-apiserver
+    template: 'jinja'
+    raises: True
+    result: 'Failed to render file path with error'
+  # 4. Raise error when the salt template is invalid or unrecognized
+  - source:
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: kube-apiserver
+    template: 'my_jinja'
+    raises: True
+    result: 'Unknown template specified'
```

Comment:
You are missing a lot of test here 
At least for coverage 
```
---------- coverage: platform linux2, python 2.7.18-final-0 ----------
Name                                         Stmts   Miss  Cover   Missing
--------------------------------------------------------------------------
salt/_modules/metalk8s_kubernetes_utils.py      60      2    97%   127, 156
```

Tests with some real templating using some context values

Reply:
Added

Reply:
```
---------- coverage: platform linux2, python 2.7.18-final-0 ----------
Name                                         Stmts   Miss  Cover   Missing
--------------------------------------------------------------------------
salt/_modules/metalk8s_kubernetes_utils.py      60      1    98%   127
```

Reply:
Now!
```
---------- coverage: platform linux2, python 2.7.17-final-0 ----------
Name                                         Stmts   Miss  Cover   Missing
--------------------------------------------------------------------------
salt/_modules/metalk8s_kubernetes_utils.py      60      0   100%
```